### PR TITLE
Doxygen Integration

### PR DESCRIPTION
**Pull request by amiclaus**
Add Doxygen to no-OS repository.

Integrate Doxygen into TravisCI to auto-generate documentation for latest version of `master` branch.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

**Comment by amiclaus** [comment]
V2:

- Fix Codacy issues

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    message(FATAL_ERROR "Doxygen not found!")
+endif()
+
+set(SOURCES_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../)
+set(DOXY_WORKING_DIR ${CMAKE_CURRENT_BINARY_DIR}/doxygen_doc)
+
+set(PROJECT_NAME "no-OS")
+
+#Source files paths to be included for documentation generation
+set(DOCUMENTED_FILES "${CMAKE_CURRENT_SOURCE_DIR}/projects_page.dox \\
```
Comment (dbogdan) [inline doc/CMakeLists.txt]: The .dox files are not present at this point - they are only added by the next commit.

File: ci/travis/doxygen.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+echo_green "Documentation was generated successfully!"
+
+############################################################################
+# If the current build is not a pull request and it is on master the 
+# documentation will be pushed to the gh-pages branch
+############################################################################
+if [[ "${TRAVIS_PULL_REQUEST}" == "false" && "${TRAVIS_BRANCH}" == "master" ]]
+then
+        pushd ${TRAVIS_BUILD_DIR}/doc
+        git clone https://amiclaus:${GITHUB_TOKEN}@github.com/analogdevicesinc/no-OS --depth 1 --branch=gh-pages doc/html &>/dev/null
```
Comment (dbogdan) [inline ci/travis/doxygen.sh]: I still don't think that's a good idea to use personal GitHub accounts for these tasks.

File: doc/CMakeLists.txt (hunk 1 above)
Comment (amiclaus) [reply doc/CMakeLists.txt]: You are right. I will swap commits.

File: ci/travis/doxygen.sh (hunk 2 above)
Comment (amiclaus) [reply ci/travis/doxygen.sh]: I added an encrypted access token for these tasks.

**Comment by amiclaus** [comment]
V3:
- add encrypted access token.

**Comment by amiclaus** [comment]
Github not showing swapped commits. I will close this PR and create a new one.

**Pull request closed without merging**